compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -14253,6 +14253,14 @@ private STNode parseListBindingPattern() {
     }
 
     private STNode parseListBindingPattern(STNode openBracket, List<STNode> bindingPatternsList) {
+        if (isEndOfListBindingPattern(peek().kind) && bindingPatternsList.size() == 0) {
```

Comment:
Do we need to check whether the size is zero?

Reply:
Yes, this block is only to handle empty case. and there are some cases in which the list is not empty and the next token is `close_bracket' which will produce the wrong results if we don't use that check.